sihl_core/lib/contract.ml

```diff
@@ -1,3 +1,34 @@
-module type EMAILTRANSPORT = sig
-  val send : Email_core.t -> (unit, string) result Lwt.t
+module Email = struct
+  module type TRANSPORT = sig
+    val send : Email_core.t -> (unit, string) result Lwt.t
+  end
+end
+
+module Migration = struct
```

Comment:
I think the migration stuff does not belong into this contract definition. Especially, repositories have no value outside services and there should be no reason to expose them outside the scope of a service. Thus, I think you cannot expose migrations like this and standardizing them is waste of time.

Reply:
Migration should probably be a type, but the registry takes first class modules so I had to wrap the data in a module. I will think about a better solution.